cmd/sriovdp/sriov-device-plugin.go

```diff
@@ -373,11 +388,17 @@ func main() {
 	}
 	glog.Infof("SRIOV Network Device Plugin registered with the Kubelet")
 
-	// Catch termination signals
+	// Catch termination signals and kubelet restart
 	select {
 	case sig := <-sigCh:
 		glog.Infof("Received signal \"%v\", shutting down.", sig)
 		sm.Stop()
 		return
+	case event := <-watcher.Events:
+		if event.Op&fsnotify.Create == fsnotify.Create {
```

Comment:
Should it be we watching for 'Create' OR 'Remove' event?

Reply:
Yes, I think basically we need to capture all the 'negative' events and exit.